provider/cloudflare.go

```diff
@@ -119,29 +126,33 @@ func NewCloudFlareProvider(domainFilter DomainFilter, zoneIDFilter ZoneIDFilter,
 		zoneIDFilter: zoneIDFilter,
 		proxied:      proxied,
 		DryRun:       dryRun,
+		PaginationOptions: cloudflare.PaginationOptions{
+			PerPage: zonesPerPage,
+		},
 	}
 	return provider, nil
 }
 
 // Zones returns the list of hosted zones.
 func (p *CloudFlareProvider) Zones() ([]cloudflare.Zone, error) {
 	result := []cloudflare.Zone{}
-
-	zones, err := p.Client.ListZones()
+	ctx := context.TODO()
+	zonesResponse, err := p.Client.ListZonesContext(ctx, cloudflare.WithPagination(p.PaginationOptions))
 	if err != nil {
 		return nil, err
 	}
+	for pages := 0; pages < zonesResponse.ResultInfo.TotalPages; pages++ {
+		for _, zone := range zonesResponse.Result {
```

Comment:
You're iterating through the same page result. If you have 20 domains and set `PerPage` to 5, for example, you're going to iterate through the same 5 domains 4 times. It will only return these 5, instead of all 20.

The above API call needs to be called in a loop.

Reply:
Yes, you're absolutely right. I already proceed with my PR and wanted to change this. However, I can't do it since the branch hook declined to push my changes. I'm currently trying to figure out why 😄

Reply:
I've just added a label "wip" so people should be aware of this.

Reply:
I pushed my changes.

Reply:
Now you're calling the first page twice. The first time to get the page count, the second time to get the results. Not really efficient. Look at my code sample above for a way to do this more efficiently.

Reply:
This looks to be resolved now. 👍